internal/services/mssql/mssql_elasticpool_resource.go

```diff
@@ -201,57 +218,60 @@ func resourceMsSqlElasticPoolCreateUpdate(d *pluginsdk.ResourceData, meta interf
 
 	log.Printf("[INFO] preparing arguments for MSSQL ElasticPool creation.")
 
-	id := parse.NewElasticPoolID(subscriptionId, d.Get("resource_group_name").(string), d.Get("server_name").(string), d.Get("name").(string))
+	id := commonids.NewSqlElasticPoolID(subscriptionId, d.Get("resource_group_name").(string), d.Get("server_name").(string), d.Get("name").(string))
 
 	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id.ResourceGroup, id.ServerName, id.Name)
+		existing, err := client.Get(ctx, id)
 		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
-				return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %s", id, err)
 			}
 		}
 
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return tf.ImportAsExistsError("azurerm_mssql_elasticpool", id.ID())
 		}
 	}
 
 	location := azure.NormalizeLocation(d.Get("location").(string))
 	sku := expandMsSqlElasticPoolSku(d)
-	t := d.Get("tags").(map[string]interface{})
+
+	// NOTE: Set the default value, if the field exists in the config the only value
+	// that it could be is 'VBS'...
+	enclaveType := elasticpools.AlwaysEncryptedEnclaveTypeDefault
+	if _, ok := d.GetOk("enclave_type"); ok {
+		enclaveType = elasticpools.AlwaysEncryptedEnclaveTypeVBS
+	}
```

Comment:
since we're setting the value from the API into the state, could we explicitly load the value from the config here? Else if we added another value for this field in the future (in the Schema) we'd miss we're hard-coding this at review time:

```suggestion
	enclaveType := elasticpools.AlwaysEncryptedEnclaveTypeDefault
	if v, ok := d.GetOk("enclave_type"); ok {
		enclaveType = elasticpools.AlwaysEncryptedEnclaveType(v.(string))
	}
```

Reply:
Fixed.